src/common/snippets/tests/src/pass/precision_propagation.cpp

```diff
@@ -0,0 +1,319 @@
+        constructor_validate_and_infer_types();
+    }
+
+    DummyAdd(const ngraph::opset1::Add& add)
+        : Add(add.get_input_source_output(0), add.get_input_source_output(1), add.get_autob()) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyAdd() = default;
+
+    void validate_and_infer_types() override {
+        const auto input_type1 = get_input_element_type(0);
+        const auto input_type2 = get_input_element_type(1);
+
+        const element::Type output_type = (input_type1 == element::i8) || (input_type2 == element::i8) ?
+            element::i32 :
+            get_input_element_type(0);
+
+        set_output_type(0, output_type, get_input_partial_shape(0));
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyAdd>(new_args.at(0), new_args.at(1), this->get_autob());
+    }
+};
+
+class DummyPrecisionPropogationTargetMachine : public DummyTargetMachine {
+public:
+    DummyPrecisionPropogationTargetMachine(
+        const std::set<std::vector<element::Type>>& op1_supported_precisions,
+        const std::set<std::vector<element::Type>>& op2_supported_precisions)
+        : DummyTargetMachine() {
+        jitters[DummyAdd::get_type_info_static()] = ngraph::snippets::jitters_value {
+            [](const std::shared_ptr<ngraph::Node>& n) { return std::make_shared<DummyEmitter>(); },
+            [&op1_supported_precisions](const std::shared_ptr<ngraph::Node>& n) { return op1_supported_precisions; }};
+        jitters[op::v1::Maximum::get_type_info_static()] = ngraph::snippets::jitters_value{
+            [](const std::shared_ptr<ngraph::Node>& n) { return std::make_shared<DummyEmitter>(); },
+            [&op2_supported_precisions](const std::shared_ptr<ngraph::Node>&n) { return op2_supported_precisions; }};
+
+        auto default_jitter = ngraph::snippets::jitters_value{
+            [](const std::shared_ptr<ngraph::Node>& n) { return std::make_shared<DummyEmitter>(); },
+            [](const std::shared_ptr<ngraph::Node>& n) { return std::set<std::vector<element::Type>>{};} };
+        jitters[ngraph::snippets::op::ConvertSaturation::get_type_info_static()] = default_jitter;
+    }
+};
+
+} // namespace
+
+std::string PrecisionPropagationTest::getTestCaseName(testing::TestParamInfo<PrecisionPropagationParams> obj) {
+    std::pair<Shape, Shape> shapes;
+    PrecisionPropagationParamsValues test_values;
+    std::tie(shapes, test_values) = obj.param;
+
+    auto to_string = [](const std::set<std::vector<element::Type>>& precisions_pack) noexcept {
+        auto to_string = [](const std::vector<element::Type>& precisions) noexcept {
+            std::ostringstream result;
+            result << "{";
+            for (const auto& precision : precisions) {
+                result << precision << "_";
+            }
+            result << "}";
+            return result.str();
+        };
```

Comment:
Maybe we can use `CommonTestUtils::vec2str` instead of custom logic? What do you think?

Reply:
done, thanks